File: csharp/806-NumbersOfLinesToWriteString.cs
// 2022-4-12
// Runtime 148ms(33.7%)
// Memory 41.5mb(36.0%)

public class Solution {
    public int[] NumberOfLines(int[] widths, string s) {
        int lines = 1, row = 0;
        int[] ret = new int[2];
        for(int i = 0; i < s.Length; i++) {
            row += widths[s[i] - 'a'];
            if(row > 100) {
                lines++;
                row = widths[s[i] - 'a'];
            }
        }
        ret[0] = lines;
        ret[1] = row;
        return ret;

    }
}

File: cpp/806-NumbersOfLinesToWriteString.cpp
// 2022-4-12
// Runtime 0ms(100.0%)
// Memory 6.5mb(96.2%)

/**
 * 96.2%啊啊啊啊啊啊啊啊啊啊100%啊啊啊啊啊啊
 */

class Solution {
public:
    vector<int> numberOfLines(vector<int>& widths, string s) {
        int lines(1), row(0);
        for(int i(0); i < s.size(); i++) {
            row += widths[s[i] - 'a'];
            if(row > 100) {
                ++lines;
                row = widths[s[i] - 'a'];
            }
        }
        return {lines, row};
    }
};
